### ISSUE TITLE
When text does not fit, TextBlock does not always correctly show ellipsis 

### ISSUE THREAD

**Issue by rickvanderheijden** [AvaloniaUI/Avalonia#13395]
## Describe the bug
When placing a long line of text in a TextBlock the ellipsis isn't correctly placed, depending on the font size. Textbox properties are set:

TextTrimming="CharacterEllipsis"
TextWrapping="Wrap"

The same setting behave correctly in Avalonia 0.10.22. Using a the most recent nightly build (1.1.999-cibuild0041289-beta) has the same behavior.

Windows display scaling is set to 100%.

## To Reproduce

Use attached zip file with sample project. Change font size settings if needed. Build and run.
[MyApp.zip](https://github.com/AvaloniaUI/Avalonia/files/13176203/MyApp.zip)

## Screenshots
Avalonia 11.0.5, font size 14:
![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/b79e34b7-f072-4a29-9853-5ed1e936272f)

Avalonia 11.0.5, font size 15:
![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/c30033bb-eec5-4fc4-8f97-1707748bf4f2)

Avalonia 0.10.22, font size 14
![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/ae00b2de-d2e9-49ee-a249-7adbcac29437)

Avalonia 0.10.22, font size 15
![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/149a795f-7896-487e-a5bc-f00529f90f7b)

## Environment 

- OS:  Windows 10 Enterprise 21H2
- Avalonia-Version: 11.0.5

**Comment by Gillibald** [comment]
I can't see the issue with Avalonia `11.0.5` What is wrong with the result?

**Comment by rickvanderheijden** [comment]
In the second screenshot (with font size 15) you can see that the ellipsis are not placed, while they are there when the font size is 14..

![image](https://github.com/AvaloniaUI/Avalonia/assets/31483005/2b44d9dd-76d2-485e-99ca-3c5f971efeff)

 Older Avalonia does show them in both font sizes

**Comment by Gillibald** [comment]
I see. So only 3 lines fit into the height and not ellipsis is shown in that case.

**Comment by rickvanderheijden** [comment]
Correct... the original text is "Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text. Some text."

**Comment by StipecMV** [comment]
Hi guys, 
I have a similar problem to @rickvanderheijden.
I tested it with 11.0.10 Avalonia and the result is the same. 
Why TextTrimming does not work in some cases together with TextWrapping @Gillibald ?

Note that the text "Some text." is written there 16 times and only 9 times does it fit into TextBlock, 
so I expect that text will be trimmed as that property is set!

**Comment by sanders-ict** [comment]
Hi all,

Just retested @rickvanderheijden 's test app with the latest Avalonia release (11.1.3) after updating the app to NET8. Still the same behavior can be seen. Any news on in which release this is planned to be fixed ?

![image](https://github.com/user-attachments/assets/0c236cb6-55dc-49f7-9b84-dee84078f56b)

**Comment by timunie** [comment]
No due date yet. You can ask for priority in case you have a support agreement.

**Comment by rent-a-developer** [comment]
I am having the same issue (Avalonia 12.0.4):
```xaml
<Border BorderBrush="Green"
        BorderThickness="1"
        HorizontalAlignment="Center"
        VerticalAlignment="Center">
    <TextBlock
        Text="_AAAAAAAAAA_ _BBBBBBBBBB_ _CCCCCCCCCC_ _DDDDDDDDDD_ _EEEEEEEEEE_ _FFFFFFFFFF_ _GGGGGGGGGG_ _HHHHHHHHHH_ _IIIIIIIIII_ _JJJJJJJJJJ_ _KKKKKKKKKK_ _LLLLLLLLLL_ _MMMMMMMMMM_ _NNNNNNNNNN_ _OOOOOOOOOO_ _PPPPPPPPPP_ _QQQQQQQQQQ_ _RRRRRRRRRR_ _SSSSSSSSSS_ _TTTTTTTTTT_ _UUUUUUUUUU_ _VVVVVVVVVV_ _WWWWWWWWWW_ _XXXXXXXXXX_ _YYYYYYYYYY_ _ZZZZZZZZZZ_"
        MaxLines="5"
        MaxWidth="350"
        TextWrapping="Wrap"
        TextTrimming="CharacterEllipsis" />
</Border>
```

<img width="597" height="224" alt="Image" src="https://github.com/user-attachments/assets/d373cf9c-727c-4d5a-8886-cdf241ecae1f" />
